src/network/serverpackethandler.cpp

```diff
@@ -1359,22 +1359,24 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 		(only when digging or placing things)
 	*/
 	if (action == 0 || action == 2 || action == 3) {
-		float d = player_pos.getDistanceFrom(pointed_pos_under);
-		float max_d = BS * 14; // Just some large enough value
-		if (d > max_d) {
-			actionstream << "Player " << player->getName()
-					<< " tried to access " << pointed.dump()
-					<< " from too far: "
-					<< "d=" << d <<", max_d=" << max_d
-					<< ". ignoring." << std::endl;
-			// Re-send block to revert change on client-side
-			RemoteClient *client = getClient(pkt->getPeerId());
-			v3s16 blockpos = getNodeBlockPos(floatToInt(pointed_pos_under, BS));
-			client->SetBlockNotSent(blockpos);
-			// Call callbacks
-			m_script->on_cheat(playersao, "interacted_too_far");
-			// Do nothing else
-			return;
+		if (!isSingleplayer() && !g_settings->getBool("disable_anticheat")) {
```

Comment:
i think the output of `(!isSingleplayer() && !g_settings->getBool("disable_anticheat"))` should be cache instead rather than the setting.
although where to put it and what to call it is the question